### PR TITLE
Add Ndiff Wrapper

### PR DESCRIPTION
**Pull request by juicebyjustin**
Hi Thom, another pull request for you. I added an Ndiff wrapper.

I included 2 XML Nmap output files for testing. Honestly, I don't know where test files should go so I placed them in a folder inside bin/debug in the test project. Let me know how you want it and I will modify to fit.

I'm using VS2017 so I don't know if the commited .csproj files will cause any issue.

### CODE REVIEW

**Comment by thomdixon** [comment]
Thanks again! I'll have to check this one out later this week (probably this weekend). I don't have access to a computer at the moment, so I can't check if the csproj file causes any issues. The code looks good. I'd like to repeat the `NmapOptions` pattern for Ndiff, though.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (thomdixon) [inline .gitignore:2]: Can you maybe add a `.gitignore` from [gitignore.io](https://www.gitignore.io/api/csharp%2Cvisualstudio) instead?

File: SaltwaterTaffy/Ndiff.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        /// <summary>
+        ///     Run the Ndiff tool to compare output files from Nmap.
+        /// </summary>
+        /// <param name="file1">Original File</param>
+        /// <param name="file2">Modified File</param>
+        /// <param name="verbose">Also show hosts and ports that haven't changed.</param>
+        /// <param name="xml">Display output in XML format.</param>
+        /// <returns>Output from Ndiff application.</returns>
+        public static string RunDiff(bool verbose, bool xml, string file1, string file2)
```
Comment (thomdixon) [inline SaltwaterTaffy/Ndiff.cs]: Would prefer to use the `NmapOptions` pattern here (maybe an `NdiffOptions` object), but I'm fine leaving it like this, if this is the only functionality you need.

File: SaltwaterTaffy/Ndiff.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+                    process.WaitForExit();
+                }
+
+
+                return string.IsNullOrEmpty(error) ? output : error;
+            }
+            catch (Exception ex)
+            {
+                throw new Exception(string.Format("Failed to run diff. Ex: {0}", ex.ToString()));
```
Comment (thomdixon) [inline SaltwaterTaffy/Ndiff.cs]: Would use [`NmapException`](https://github.com/thomdixon/SaltwaterTaffy/blob/master/SaltwaterTaffy/Nmap.cs#L505) here instead.

File: SaltwaterTaffy/Ndiff.cs (hunk 2 above)
Comment (thomdixon) [inline SaltwaterTaffy/Ndiff.cs]: Why not name this `Run`?

**Review by thomdixon — CHANGES_REQUESTED** [review]
I have a few comments that I'd like your input on, but otherwise looks good to me.

File: SaltwaterTaffy/Ndiff.cs (hunk 3 above)
Comment (juicebyjustin) [reply SaltwaterTaffy/Ndiff.cs]: Certainly.

File: SaltwaterTaffy/Ndiff.cs (hunk 2 above)
Comment (juicebyjustin) [reply SaltwaterTaffy/Ndiff.cs]: I honestly do not know. I'll change to Run.

File: .gitignore (hunk 1 above)
Comment (juicebyjustin) [reply .gitignore:2]: Will do.

File: SaltwaterTaffy/Ndiff.cs (hunk 2 above)
Comment (juicebyjustin) [reply SaltwaterTaffy/Ndiff.cs]: That's make sense to use the NmapOptions pattern. I'll try to have another commit before the weekend with this and the other changes per your comments.

**Comment by jmooney5115** [comment]
Hi. The changes were easier than expected. I'm new to pull requests. How do add the modified code here? I committed it to the same 'ndiff-addition' branch.

**Review by thomdixon — APPROVED** [review]
Looks great! Thanks so much!

**Comment by juicebyjustin** [comment]
You're welcome, I'm glad to help.

**Pull request merged**